test/Microsoft.Azure.WebJobs.Extensions.EventHubs.Tests/EventHubApplicationInsightsTest.cs

```diff
@@ -0,0 +1,438 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.Azure.EventHubs;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests
+{
+    public class EventHubApplicationInsightsTests : IDisposable
+    {
+        private const string TestHubName = "webjobstesthub";
+        private const int Timeout = 30000;
+        private static EventWaitHandle _eventWait;
+        private readonly TestTelemetryChannel _channel = new TestTelemetryChannel();
+        private static string _testPrefix;
+        private readonly string _connection;
+        private readonly string _endpoint;
+        private readonly EventHubsConnectionStringBuilder _connectionBuilder;
+        private readonly JsonSerializerSettings jsonSettingThrowOnError = new JsonSerializerSettings
+        {
+            MissingMemberHandling = MissingMemberHandling.Error,
+            ReferenceLoopHandling = ReferenceLoopHandling.Error,
+            NullValueHandling = NullValueHandling.Include,
+            DefaultValueHandling = DefaultValueHandling.Include,
+        };
+
+        public EventHubApplicationInsightsTests()
+        {
+            _eventWait = new ManualResetEvent(initialState: false);
+            _testPrefix = Guid.NewGuid().ToString();
+
+            var config = new ConfigurationBuilder()
+                .AddEnvironmentVariables()
+                .AddTestSettings()
+                .Build();
+
+            _connection = config.GetConnectionStringOrSetting("AzureWebJobsTestHubConnection");
+            Assert.True(!string.IsNullOrEmpty(_connection), "Required test connection string is missing.");
```

Comment:
nit -- could you put the name of the missing connection string here? I don't think we do that elsewhere, but it'd be good to start.